LEAF_Request_Portal/js/form.js

```diff
@@ -6,272 +6,349 @@ var formValidator = {};
-		$('#' + dialog.btnSaveID).empty().html('<img src="images/indicator.gif" alt="saving" /> Saving...');
-
-		$('#' + htmlFormID).find(':input:disabled').removeAttr('disabled');
-		var data = {recordID: recordID};
-		$('#' + htmlFormID).serializeArray().map(function(x) {
-			if (x.name.includes('_multiselect')) {
-				const i = x.name.indexOf('_multiselect');
-				if (x.value === '') { //selected if no options are chosen
-					data[x.name.slice(0, i)] = x.value;
-				} else {
-					data[x.name.slice(0, i)] ? data[x.name.slice(0, i)].push(x.value) : data[x.name.slice(0, i)] = [x.value];
-				}
-			} else data[x.name] = x.value;
-		});
-
-		if(hasTable){
+        return input;
+    }
+
+    
+    function handleConditionalIndicators(formConditions) {
+        const allowedChildFormats = ['dropdown', 'text', 'multiselect'];
+        const formConditionsByChild = formConditions;
+        let currentChildInfo = {};
+        
+        const checkConditions = (event, selected, parID=0)=> {
+            const parentElID = event !== null ? parseInt(event.target.id) : parseInt(parID);
+
+            const linkedParentConditions = getConditionsLinkedToParent(parentElID);
+            let uniqueChildIDs = linkedParentConditions.map(c => c.childIndID);
+            uniqueChildIDs = Array.from(new Set(uniqueChildIDs));
+
+            let linkedChildConditions = [];
+            uniqueChildIDs.forEach(id => {
+                linkedChildConditions.push(...getConditionsLinkedToChild(id, parentElID));
+            });
+
+            const allConditions = [...linkedParentConditions, ...linkedChildConditions];
+
+            const conditionsByChild = {}
+            allConditions.map(c => {
+                conditionsByChild[c.childIndID] ? conditionsByChild[c.childIndID].push(c) : conditionsByChild[c.childIndID] = [c];
+            })
+
+            setTimeout(() => {  //some multiselect updates don't work unless the stack is cleared
+                for (let childID in conditionsByChild) {
+                    makeComparisons(childID, conditionsByChild[childID]);
+                }
+            }, 0);
+        }
+
+        const getConditionsLinkedToParent = (parentID)=> {
+            let conditionsLinkedToParent = [];
+            for (let entry in formConditionsByChild) {
+                const formConditions = formConditionsByChild[entry].conditions || [];
+                formConditions.forEach(c => {
+                    const formatIsEnabled = allowedChildFormats.some(f => f === c.childFormat);
+                    //do not include conditions if the recorded condition format (condition.childFormat) does not
+                    //match the current format, as this would have unpredictable results
+                    if (formConditionsByChild[entry].format === c.childFormat && 
+                        formatIsEnabled &&
+                        parseInt(c.parentIndID) === parseInt(parentID)) {
+                        conditionsLinkedToParent.push({...c});
```

Comment:
Couldn't that just be this?:
```suggestion
                        conditionsLinkedToParent.push(c);
```